server/service/src/invoice/inventory_adjustment/insert.rs

```diff
@@ -0,0 +1,245 @@
+                ctx,
+                ActivityLogType::InventoryAdjustment,
+                Some(verified_invoice.id.to_owned()),
+                None,
+                None,
+            )?;
+
+            get_invoice(ctx, None, &verified_invoice.id)
+                .map_err(InsertInventoryAdjustmentError::DatabaseError)?
+                .ok_or(InsertInventoryAdjustmentError::NewlyCreatedInvoiceDoesNotExist)
+        })
+        .map_err(|error| error.to_inner_error())?;
+
+    Ok(invoice)
+}
+
+impl From<RepositoryError> for InsertInventoryAdjustmentError {
+    fn from(error: RepositoryError) -> Self {
+        InsertInventoryAdjustmentError::DatabaseError(error)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use repository::{
+        mock::{
+            mock_stock_line_a, mock_store_a, mock_store_b, mock_user_account_a, MockDataInserts,
+        },
+        test_db::setup_all,
+        InvoiceRowRepository, InvoiceRowStatus, StockLine,
+    };
+    use util::inline_edit;
+
+    use crate::{
+        invoice::inventory_adjustment::insert::InsertInventoryAdjustment,
+        service_provider::ServiceProvider,
+    };
+
+    use super::InsertInventoryAdjustmentError;
+
+    type ServiceError = InsertInventoryAdjustmentError;
+
+    #[actix_rt::test]
+    async fn insert_inventory_adjustment_errors() {
+        let (_, _, connection_manager, _) =
+            setup_all("insert_inventory_adjustment_errors", MockDataInserts::all()).await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let mut context = service_provider
+            .context(mock_store_a().id, "".to_string())
+            .unwrap();
+        let service = service_provider.invoice_service;
+
+        // Stockline doesn't exist
+        assert_eq!(
+            service.insert_inventory_adjustment(
+                &context,
+                InsertInventoryAdjustment {
+                    stock_line_id: "x".to_string(),
+                    ..Default::default()
+                }
+            ),
+            Err(ServiceError::StockLineDoesNotExist)
+        );
+
+        // Wrong store
+        context.store_id = mock_store_b().id;
+        assert_eq!(
+            service.insert_inventory_adjustment(
+                &context,
+                InsertInventoryAdjustment {
+                    stock_line_id: mock_stock_line_a().id,
+                    ..Default::default()
+                }
+            ),
+            Err(ServiceError::InvalidStore)
+        );
+        context.store_id = mock_store_a().id;
+
+        // Missing reason
+        // assert_eq!(
+        //     service.insert_inventory_adjustment(
+        //         &context,
+        //         InsertInventoryAdjustment {
+        //             stock_line_id: mock_stock_line_a().id,
+        //             ..Default::default()
+        //         }
+        //     ),
+        //     Err(ServiceError::AdjustmentReasonNotProvided)
+        // );
```

Comment:
There seems to be a problem with `check_active_adjustment_reasons` that is throwing an error causing this test to fail.

Reply:
I'm not seeing an error thrown? Just that this check is being skipped - which is expected behaviour (no reasons are configured in the mock data by default, and if no reasons configured, we don't expect a reason id to be provided.) Here's how I would set up for this test: https://github.com/msupply-foundation/open-msupply/compare/3498-adjustment-back-end...3498-adjustment-back-end(test-inventory-adjustment-reason-not-provided)

Reply:
Thanks! 3821e3d21